### PR TITLE
Add packages: Gnu Guile, Guile Json

### PR DESCRIPTION
**Pull request by o-nly**
Towards the goal of packaging Gnu Guix

### CODE REVIEW

File: packages/guile-xyz/guile-json/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/aconchillo/guile-json
+TERMUX_PKG_DESCRIPTION="Guile-JSON supports parsing and building JSON documents according to the specification(http://json.org)."
+TERMUX_PKG_LICENSE="GPL3 or later"
```
Comment (xeffyr) [inline packages/guile-xyz/guile-json/build.sh]: License names cannot be arbitrary. License name should be one of: AFL-2.1, AFL-3.0, AGPL-V3, Apache-1.0, Apache-1.1, Apache-2.0, APL-1.0, APSL-2.0, Artistic-License-2.0, Attribution, Bouncy-Castle, BSD, BSD 2-Clause, BSD 3-Clause, BSL-1.0, CA-TOSL-1.1, CC0-1.0, CDDL-1.0, Codehaus, CPAL-1.0, CPL-1.0, CPOL-1.02, CUAOFFICE-1.0, Day, Day-Addendum, ECL2, Eiffel-2.0, Entessa-1.0, EPL-1.0, EPL-2.0, EUDATAGRID, EUPL-1.1, Fair, Facebook-Platform, Frameworx-1.0, Go, GPL-2.0, GPL-2.0+CE, GPL-3.0, Historical, HSQLDB, IBMPL-1.0, ImageMagick, IPAFont-1.0, ISC, IU-Extreme-1.1.1, JA-SIG, JSON, JTidy, LGPL-2.1, LGPL-3.0, Libpng, LPPL-1.0, Lucent-1.02, MirOS, MIT, Motosoto-0.9.1, Mozilla-1.1, MPL-2.0, MS-PL, MS-RL, Multics, NASA-1.3, NAUMEN, Nethack, Nokia-1.0a, NOSL-3.0, NTP, NUnit-2.6.3, NUnit-Test-Adapter-2.6.3, OCLC-2.0, Openfont-1.1, Opengroup, OpenSSL, OSL-3.0, PHP-3.0, PostgreSQL, Public Domain, Public Domain - SUN, PythonPL, PythonSoftFoundation, QTPL-1.0, Real-1.0, RicohPL, RPL-1.5, Scala, SimPL-2.0, Sleepycat, SUNPublic-1.0, Sybase-1.0, TMate, Unlicense, UoI-NCSA, VIM License, VovidaPL-1.0, W3C, WTFPL, wxWindows, Xnet, ZLIB, ZPL-2.0.

See https://www.jfrog.com/confluence/display/BT/Bintray+REST+API for information.

File: packages/guile-xyz/guile-json/build.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+TERMUX_PKG_VERSION=3.1.0
+TERMUX_PKG_NAME=guile-json
```
Comment (xeffyr) [inline packages/guile-xyz/guile-json/build.sh]: TERMUX_PKG_NAME should not be specified.
Package name are determined by package directory name.

File: packages/guile-xyz/guile-json/build.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+TERMUX_PKG_REVISION=1
```
Comment (xeffyr) [inline packages/guile-xyz/guile-json/build.sh]: Revision should not be specified for initial release.

File: packages/guile-xyz/guile-json/build.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+TERMUX_PKG_SHA256=e303d0db3f34ac515b8f61db8ed35be8b6c5014d7ab4a5a8ef9b44723ed3d8f9
+TERMUX_PKG_SRCURL=https://download.savannah.nongnu.org/releases/${TERMUX_PKG_NAME}/${TERMUX_PKG_NAME}-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_DEPENDS="guile pkg-config"
```
Comment (xeffyr) [inline packages/guile-xyz/guile-json/build.sh]: Dependency lists **must** have comma as delimiter:
```Shell
TERMUX_PKG_DEPENDS="guile, pkg-config"
```

File: packages/guile/build.sh [hunk 5]
```diff
@@ -0,0 +1,9 @@
+TERMUX_PKG_HOMEPAGE=https://www.gnu.org/software/guile/
+TERMUX_PKG_DESCRIPTION="Guile is the official extension language of the GNU system."
+TERMUX_PKG_LICENSE="LGPL-3.0 or later"
+TERMUX_PKG_VERSION=2.2.4
+TERMUX_PKG_REVISION=1
+TERMUX_PKG_SHA256=33b904c0bf4e48e156f3fb1d0e6b0392033bd610c6c9d9a0410c6e0ea96a3e5c
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/guile/guile-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_DEPENDS="libffi libunistring libtool libgc libgmp"
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS=""
```
Comment (xeffyr) [inline packages/guile/build.sh]: Same problems as mentioned in my previous comments.

File: packages/guile-xyz/guile-json/build.sh (hunk 4 above)
Comment (xeffyr) [reply packages/guile-xyz/guile-json/build.sh]: TERMUX_PKG_EXTRA_CONFIGURE_ARGS should not be specified if empty.

**Comment by xeffyr** [comment]
>  packages/guile-xyz/guile-json/build.sh

Packages cannot be located in arbitrary directories.

**Comment by xeffyr** [comment]
Fixed everything by myself. But that doesn't mean that guile suddenly became available.

**Comment by o-nly** [comment]
Checks have failed(https://travis-ci.org/termux/termux-packages/jobs/501093107)

...
checking for libltdl... no
configure: error: GNU libltdl (Libtool) not found, see README.
Input/output error @ io_fillbuf - fd:9 /dev/pts/1
Error building guile-json
The command "bundle exec rake build["${ARGS}"]" exited with 1.

Guile-json requires pkg-config, so i'd guess it should fail. But it fails for other reasons^. It can't find `libltdl`. I see now that instead of `libtool`, guile should depend on `libtldl` sub-package. (https://github.com/termux/termux-packages/blob/master/packages/libtool/libltdl.subpackage.sh)

**Comment by xeffyr** [comment]
See https://github.com/termux/termux-packages/pull/1604/files

**Comment by xeffyr** [comment]
And this (your build):
```
  SNARF  variable.doc
  SNARF  vectors.doc
  SNARF  version.doc
  SNARF  vports.doc
  SNARF  weak-set.doc
  SNARF  weak-table.doc
  SNARF  weak-vector.doc
  SNARF  posix.doc
  SNARF  net_db.doc
  SNARF  socket.doc
  SNARF  regex-posix.doc
  CCLD     libguile-2.2.la
  CCLD     guile
./.libs/libguile-2.2.so: error: undefined reference to 'csqrt'
clang80: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [Makefile:2564: guile] Error 1
make[3]: Leaving directory '/home/builder/.termux-build/guile/build/libguile'
make[2]: *** [Makefile:2423: all] Error 2
make[2]: Leaving directory '/home/builder/.termux-build/guile/build/libguile'
make[1]: *** [Makefile:1857: all-recursive] Error 1
make[1]: Leaving directory '/home/builder/.termux-build/guile/build'
make: *** [Makefile:1743: all] Error 2
```

**Comment by xeffyr** [comment]
Also, building guile on device is not same as **cross-compiling** it with Termux build system.

**Comment by xeffyr** [comment]
> checking for guile... not-found
configure: error: a native Guile 2.2.4 is required to cross-build Guile

Enough to say that current variant of build scripts is completely unsuitable.

**Comment by o-nly** [comment]
Thanks, that seems reasonable.

**Pull request closed without merging**